src/tools/miri/src/diagnostics.rs

```diff
@@ -336,15 +336,11 @@ pub fn report_error<'tcx, 'mir>(
 
     // FIXME(fee1-dead), HACK: we want to use the error as title therefore we can just extract the
```

Comment:
@davidtwco const_eval errors are used here as titles, that is why we need to eagerly translate.

We are also doing it to avoid args "pollution" in the DiagnosticBuilder.

cc: @fee1-dead

Reply:
Even if you use the derive macro, the hack that worked before still works:

```rust
    let e = {
        let handler = &ecx.tcx.sess.parse_sess.span_diagnostic;
        let mut diag = ecx.tcx.sess.struct_allow("");
        let msg = e.diagnostic_message();
        e.add_args(handler, &mut diag);
        let s = handler.eagerly_translate_to_string(msg, diag.args());
        diag.cancel();
        s
    };
```

the FIXME is just there to encourage someone to find a better way than creating a dummy diagnostic to extract the title, and doesn't have to be fixed by this PR.

Reply:
Unless I'm still misunderstanding something, the approach I've suggested should work for this - because the `InterpErrorExt` will call `IntoDiagnostic` on `InterpError`, it'll get the `InterpError`'s primary message, and if it is the opposite of that which you want, then you can always change it in `InterpErrorExt`'s implementation.